qcodes/instrument/parameter.py

```diff
@@ -421,6 +422,65 @@ def snapshot_base(self, update: bool = True,
 
         return state
 
+    def _from_value_to_raw_value(self, value: ParamDataType
+                                 ) -> ParamRawDataType:
+        raw_value: ParamRawDataType
+
+        # Convert set values using val_mapping dictionary
+        if self.val_mapping is not None:
+            raw_value = self.val_mapping[value]
+        else:
+            raw_value = value
+
+        # transverse transformation in reverse order as compared to
+        # getter: apply scale first
+        if self.scale is not None:
+            if isinstance(self.scale, collections.abc.Iterable):
+                # Scale contains multiple elements, one for each value
+                raw_value = tuple(val * scale for val, scale
+                                  in zip(raw_value, self.scale))
+            else:
+                # Use single scale for all values
+                raw_value *= self.scale
+
+        # apply offset next
+        if self.offset is not None:
+            if isinstance(self.offset, collections.abc.Iterable):
+                # offset contains multiple elements, one for each value
+                raw_value = tuple(val + offset for val, offset
+                                  in zip(raw_value, self.offset))
+            else:
+                # Use single offset for all values
+                raw_value += self.offset
+
+        # parser last
+        if self.set_parser is not None:
+            raw_value = self.set_parser(raw_value)
+
+        return raw_value
+
+    def set_cached(self, value: ParamDataType) -> None:
+        """
+        Set the cached value of the parameter without invoking the
+        ``set_cmd`` of the parameter (if it has one). For example, in case of
+        an instrument parameter, calling :meth:`set_cached` as opposed to
+        calling :meth:`set` will only change the internally-stored value of
```

Comment:
Docs build fail since you cannot put in a reference to set that is dynamically generated